project_generator/tools/tool.py

```diff
@@ -103,3 +106,52 @@ def fixup_executable(self, exe_path):
     @staticmethod
     def is_supported_by_default(target):
         return False
+
+    def _expand_data(self, old_data, new_data, group):
+        """ data expansion - uvision needs filename and path separately. """
+        for file in sum(old_data.values(), []):
+            if file:
+                extension = file.split(".")[-1].lower()
+                if extension in self.file_types.keys():
```

Comment:
This handles the debug message when file type is not recognized. I tried to make sure that nothing changed in the implementation.